File: app/src/main/java/wseemann/media/romote/activity/DeviceInfoActivity.java
package wseemann.media.romote.activity;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.fragment.app.FragmentTransaction;

import wseemann.media.romote.R;
import wseemann.media.romote.fragment.DeviceInfoFragment;

/**
 * Created by wseemann on 6/19/16.
 */
public class DeviceInfoActivity extends ConnectivityActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_deviceinfo);

        String serialNumber = getIntent().getStringExtra("serial_number");
        String host = getIntent().getStringExtra("host");

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.add(android.R.id.content, DeviceInfoFragment.getInstance(serialNumber, host)).commit();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
            default:
                break;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/wseemann/media/romote/activity/DeviceInfoActivity.kt
package wseemann.media.romote.activity

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.compose.runtime.getValue
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import dagger.hilt.android.AndroidEntryPoint
import wseemann.media.romote.composables.DeviceInfoScreen
import wseemann.media.romote.composables.theme.RomoteTheme
import wseemann.media.romote.viewmodels.DeviceInfoScreenViewModel

/**
 * Everything the device reports about itself. Reached from the device lists, either for a paired
 * device (by serial number) or one that was just discovered (by host); both arrive as Intent
 * extras, which DeviceInfoScreenViewModel reads from its SavedStateHandle.
 */
@AndroidEntryPoint
class DeviceInfoActivity : ConnectivityActivity() {

    private val deviceInfoScreenViewModel: DeviceInfoScreenViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContent {
            val uiState by deviceInfoScreenViewModel.uiState.collectAsStateWithLifecycle()

            RomoteTheme {
                DeviceInfoScreen(uiState = uiState, onBackClick = { finish() })
            }
        }
    }
}
